x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecksTests.java

```diff
@@ -60,4 +62,48 @@ public void testCheckProcessors() {
         assertSettingDeprecationsAndWarnings(new Setting<?>[]{EsExecutors.PROCESSORS_SETTING});
     }
 
+    public void testCheckMissingRealmOrders() {
+        String realmName = randomAlphaOfLengthBetween(4, 12);
+        String realmType = randomAlphaOfLengthBetween(4, 12);
+        final Settings settings =
+            Settings.builder()
+                .put("xpack.security.authc.realms." + realmType + "." + realmName + ".enabled", "true").build();
+
+        final PluginsAndModules pluginsAndModules = new PluginsAndModules(Collections.emptyList(), Collections.emptyList());
+        final List<DeprecationIssue> deprecationIssues =
+            DeprecationChecks.filterChecks(DeprecationChecks.NODE_SETTINGS_CHECKS, c -> c.apply(settings, pluginsAndModules));
+
+        assertEquals(1, deprecationIssues.size());
+        assertEquals(new DeprecationIssue(
+            DeprecationIssue.Level.CRITICAL,
+            "Realm order will be required in next major release.",
+            "",
+            String.format(
+                Locale.ROOT,
+                "Found realms without order config: [%s]. In next major release, node will fail to start with missing realm order",
+                realmName
+            )
+        ), deprecationIssues.get(0));
+    }
+
+    public void testCheckUniqueRealmOrders() {
+        final int order = randomInt();
+        final Settings settings = Settings.builder()
+            .put("xpack.security.authc.realms."
+                + randomAlphaOfLengthBetween(4, 12) + "." + randomAlphaOfLengthBetween(4, 12) + ".order", order)
+            .put("xpack.security.authc.realms." 
+                + randomAlphaOfLengthBetween(4, 12) + "." + randomAlphaOfLengthBetween(4, 12) + ".order", order)
```

Comment:
Can we add another realm with `order + 1` and verify that it isn't reported.

Reply:
Yes we should. The test cases are not exhaustive. I have updated them to have better coverage.